Sources/KlaviyoSwift/FileUtils.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  FileUtils.swift
+//  KlaviyoSwift
+//
+//  Created by Noah Durell on 9/26/22.
+//
+
+import Foundation
+
+func write(data: Data, url: URL) throws {
+    try data.write(to: url, options: .atomic)
+}
+
+struct FileClient {
+    static let production = FileClient(
+        write: write(data:url:),
+        fileExists: FileManager.default.fileExists(atPath:),
+        removeItem: FileManager.default.removeItem(atPath:),
+        libraryDirectory: { FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask).first! }
+    )
+    var write: (Data, URL) throws -> Void
+    var fileExists: (String) -> Bool
+    var removeItem: (String) throws -> Void
+    var libraryDirectory: () -> URL
+}
+
+/**
+filePathForData: returns a string representing the filepath where archived event queues are stored
+
+- Parameter apiKey: api key for distinguishing between sets of data
+- Parameter data: name representing the event queue to locate (will be either people or events)
+- Returns: filePath string representing the file location
+*/
+func filePathForData(apiKey: String, data: String) -> URL {
+    let fileName = "klaviyo-\(apiKey)-\(data).plist"
```

Comment:
hmm thoughts on using "api key" here?

Reply:
So I'm ok with it...what were you thinking?